### PR TITLE
Fix hit lighting misalignment on argon skin with upscroll

### LINKED ISSUE #22979: osu!mania light hit with upscroll is missplaced
### Type

Cosmetic

### Bug description

Light hit in osu!mania on argo skin with upscroll enabled is missplaced

### Screenshots or videos

Upscroll:
image
Downscroll (how it should be):
image

### Version

2023.326.1

### Logs

[updater.log](https://github.com/ppy/osu/files/11072720/updater.log)
[runtime.log](https://github.com/ppy/osu/files/11072721/runtime.log)
[performance.log](https://github.com/ppy/osu/files/11072722/performance.log)
[database.log](https://github.com/ppy/osu/files/11072723/database.log)
[input.log](https://github.com/ppy/osu/files/11072724/input.log)
[network.log](https://github.com/ppy/osu/files/11072725/network.log)

### PR DESCRIPTION
**Pull request by bdach**
Closes https://github.com/ppy/osu/issues/22979

Before:

https://user-images.githubusercontent.com/20418176/228014687-b8cb5794-b8d1-47ab-a69f-56317387cace.mp4

After:

https://user-images.githubusercontent.com/20418176/228014747-bec79ef5-45e1-4822-bc5d-4f9a7fd16463.mp4

Also includes a few extra test steps in `TestSceneManiaPlayer` to cover downscroll/upscroll, because none of the skinning test scenes were really any help in spotting this. Maybe this way I'll remember to check upscroll next time...

### CODE REVIEW

File: osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs [hunk 1]
```diff
@@ -78,16 +76,8 @@ private void load(IScrollingInfo scrollingInfo)
...
-            {
-                Anchor = Anchor.TopCentre;
-                Y = ArgonNotePiece.NOTE_HEIGHT / 2;
-            }
-            else
-            {
-                Anchor = Anchor.BottomCentre;
-                Y = -ArgonNotePiece.NOTE_HEIGHT / 2;
-            }
+            Anchor = largeFaint.Anchor = largeFaint.Origin = direction.NewValue == ScrollingDirection.Up ? Anchor.TopCentre : Anchor.BottomCentre;
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs:79]: Personally I find the old code 100% more legible. Any reason for changing the code style?

Comment (bdach) [reply osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs:79]: I didn't like that with the previous code I had to repeat the compound assignment. Seemed error prone, but I'll revert to the old style if you think that's best.

Comment (peppy) [reply osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs:79]: You mean the assignment to the three different drawables? I actually didn't even see that when reading this due to the single-line formatting so I think that's another good reason to split things out 😅.

I'd even be okay with

```csharp
diff --git a/osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs b/osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs
index fb38b536dc..d490d3f944 100644
--- a/osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs
+++ b/osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs
@@ -76,8 +76,20 @@ private void load(IScrollingInfo scrollingInfo)
 
         private void onDirectionChanged(ValueChangedEvent<ScrollingDirection> direction)
         {
-            Anchor = largeFaint.Anchor = largeFaint.Origin = direction.NewValue == ScrollingDirection.Up ? Anchor.TopCentre : Anchor.BottomCentre;
-            Y = direction.NewValue == ScrollingDirection.Up ? ArgonNotePiece.NOTE_HEIGHT / 2 : -ArgonNotePiece.NOTE_HEIGHT / 2;
+            if (direction.NewValue == ScrollingDirection.Up)
+            {
+                Anchor = Anchor.TopCentre;
+                largeFaint.Anchor = Anchor.TopCentre;
+                largeFaint.Origin = Anchor.TopCentre;
+                Y = ArgonNotePiece.NOTE_HEIGHT / 2;
+            }
+            else
+            {
+                Anchor = Anchor.BottomCentre;
+                largeFaint.Anchor = Anchor.BottomCentre;
+                largeFaint.Origin = Anchor.BottomCentre;
+                Y = -ArgonNotePiece.NOTE_HEIGHT / 2;
+            }
         }
 
         public void Animate(JudgementResult result)

```

Comment (bdach) [reply osu.Game.Rulesets.Mania/Skinning/Argon/ArgonHitExplosion.cs:79]: Sure, I can apply as proposed instead.

**Review by peppy — APPROVED** [review]

**Pull request merged**